### PR TITLE
Remove model name check for REST models

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

- Remove model name check for REST models
- Add new OpenAI registry functions for GPT-3.5 and GPT-4 without restrictions for model names.

Closes https://github.com/explosion/spacy-llm/issues/351. Note that new, `Literal`-less registry functions for other REST models still have to be added.

### Corresponding documentation PR

https://github.com/explosion/spaCy/pull/13119

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran all tests in `tests` and `usage_examples/tests`, and all new and existing tests passed. This includes
  - all external tests (i. e. `pytest` ran with `--external`)
  - all tests requiring a GPU (i. e. `pytest` ran with `--gpu`)
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Review by svlandeg — COMMENTED** [review]
Looks good to me - thanks for the quick turnover!

I cancelled the external tests because 1) OpenAI API is down and 2) the tests don't actually cover the new model names, right? I guess that would be overkill, to test every combination.

Nevertheless, I'd like us to hold off a bit on merging. If OpenAI's API comes back online, I'd like to just do a quick test locally.

**Comment by rmitsch** [comment]
> I cancelled the external tests because 1) OpenAI API is down and 2) the tests don't actually cover the new model names, right? I guess that would be overkill, to test every combination.

Yes, that applies to some older models as well. We are at a point where we don't test every model. We should discuss whether that's still something we want to do.

File: spacy_llm/models/rest/openai/registry.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 def openai_gpt_4_v2(
     config: Dict[Any, Any] = SimpleFrozenDict(temperature=_DEFAULT_TEMPERATURE),
     name: Literal[
-        "gpt-4", "gpt-4-0314", "gpt-4-32k", "gpt-4-32k-0314"
+        "gpt-4", "gpt-4-0314", "gpt-4-32k", "gpt-4-32k-0314", "gpt-4-1106-preview"
```
Comment (svlandeg) [inline spacy_llm/models/rest/openai/registry.py:28]: Ugh I'm again wondering whether we should version bump `GPT-4` for this. The issue is this: we'll document that `"spacy.GPT-4.v2"` has this expanded literal, but users will actually have to upgrade spaCy for this model name to become available, right? So there will be two versions of `"spacy.GPT-4.v2"` that don't behave exactly the same.

File: spacy_llm/models/rest/openai/registry.py [hunk 2]
```diff
@@ -35,7 +64,8 @@ def openai_gpt_4_v2(
     """Returns OpenAI instance for 'gpt-4' model using REST to prompt API.
 
     config (Dict[Any, Any]): LLM config passed on to the model's initialization.
-    name (Optional[Literal["0314", "32k", "32k-0314"]]): Model to use. Base 'gpt-4' model by default.
+    name (Optional[Literal["gpt-4", "gpt-4-0314", "gpt-4-32k", "gpt-4-32k-0314"]]): Model to use. Base 'gpt-4' model by
```
Comment (svlandeg) [inline spacy_llm/models/rest/openai/registry.py:67]: TODO: update

File: spacy_llm/models/rest/openai/registry.py [hunk 3]
```diff
@@ -90,6 +152,7 @@ def openai_gpt_3_5_v2(
         "gpt-3.5-turbo-0613",
         "gpt-3.5-turbo-0613-16k",
         "gpt-3.5-turbo-instruct",
+        "gpt-3.5-turbo-1106",
```
Comment (svlandeg) [inline spacy_llm/models/rest/openai/registry.py:155]:
```suggestion
```

File: spacy_llm/models/rest/openai/registry.py [hunk 4]
```diff
@@ -100,7 +163,8 @@ def openai_gpt_3_5_v2(
 
     config (Dict[Any, Any]): LLM config passed on to the model's initialization.
     name (Literal[
-        "gpt-3.5-turbo", "gpt-3.5-turbo-16k", "gpt-3.5-turbo-0613", "gpt-3.5-turbo-0613-16k", "gpt-3.5-turbo-instruct"
+        "gpt-3.5-turbo", "gpt-3.5-turbo-16k", "gpt-3.5-turbo-0613", "gpt-3.5-turbo-0613-16k", "gpt-3.5-turbo-instruct",
+        "gpt-3.5-turbo-1106"
```
Comment (svlandeg) [inline spacy_llm/models/rest/openai/registry.py:167]:
```suggestion
        "gpt-3.5-turbo", "gpt-3.5-turbo-16k", "gpt-3.5-turbo-0613", "gpt-3.5-turbo-0613-16k", "gpt-3.5-turbo-instruct"
```

File: spacy_llm/tests/test_combinations.py [hunk 5]
```diff
@@ -12,7 +12,7 @@
 @pytest.mark.skipif(has_langchain is False, reason="LangChain is not installed")
 @pytest.mark.parametrize(
     "model",
-    ["langchain.OpenAI.v1", "spacy.GPT-3-5.v1", "spacy.GPT-3-5.v2"],
+    ["langchain.OpenAI.v1", "spacy.GPT-3-5.v2", "spacy.GPT-4.v2"],
```
Comment (svlandeg) [inline spacy_llm/tests/test_combinations.py:15]:
```suggestion
    ["langchain.OpenAI.v1", "spacy.GPT-3-5.v3", "spacy.GPT-4.v3"],
```

**Review by svlandeg — COMMENTED** [review]
Few small suggestions.

**Review by svlandeg — APPROVED** [review]

**Pull request merged**